client1x/src/main.rs

```diff
@@ -14,33 +14,60 @@ async fn main() {
 
     let start = Instant::now();
 
-    for _ in 0..200 {
-        // reusing the connection does not seem to work, not sure why
-        // sometimes panics because ECANCELED
-
-        let stream = TcpStream::connect(format!(
-            "{}:{}",
-            endpoint.host().unwrap(),
-            endpoint.port_u16().unwrap_or(80)
-        ))
-        .await
-        .unwrap();
-
-        let io = TokioIo::new(stream);
-        let (mut sender, conn) = hyper::client::conn::http1::handshake(io).await.unwrap();
-        tokio::task::spawn(async move {
-            if let Err(err) = conn.await {
-                println!("Connection failed: {:?}", err);
-            }
-        });
+    println!("Using path: {}", endpoint.path());
+
+    let stream = TcpStream::connect(format!(
+        "{}:{}",
+        endpoint.host().unwrap(),
+        endpoint.port_u16().unwrap_or(80)
+    ))
+    .await
+    .unwrap();
+    stream.set_nodelay(true).unwrap();
+    let io = TokioIo::new(stream);
+    let (mut sender, conn) = hyper::client::conn::http1::handshake(io).await.unwrap();
+
+    tokio::task::spawn(async move {
+        if let Err(err) = conn.await {
+            println!("Connection failed: {:?}", err);
+        }
+    });
```

Comment:
This moves client / socket creation out of the loop and enables the connection to be reused across requests